From 7befaa4c7b3c396df807d88ff8c61d003a28e930 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Tue, 15 Mar 2022 12:05:57 -0700
Subject: [PATCH 6/8] Respect the shared flag

emcc normally discards the -shared flag and builds a executable instead.
Generally, though, we mean it when we say -shared. This makes emcc respect
the -shared flag (by setting SIDE_MODULE=1 and removing the associated warning)
---
 emcc.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/emcc.py b/emcc.py
index 5653470dd..dc8445016 100755
--- a/emcc.py
+++ b/emcc.py
@@ -988,6 +988,9 @@ There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR P
 
     options, settings_changes, user_js_defines, newargs = parse_args(newargs)
 
+    if options.shared:
+      shared.Settings.SIDE_MODULE = 1
+
     if options.post_link or options.oformat == OFormat.BARE:
       diagnostics.warning('experimental', '--oformat=base/--post-link are experimental and subject to change.')
 
@@ -1471,10 +1474,6 @@ There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR P
       # of LTO.
       if final_suffix in EXECUTABLE_ENDINGS:
         diagnostics.warning('emcc', '-shared/-r used with executable output suffix. This behaviour is deprecated.  Please remove -shared/-r to build an executable or avoid the executable suffix (%s) when building object files.' % final_suffix)
-      else:
-        if options.shared:
-          diagnostics.warning('emcc', 'linking a library with `-shared` will emit a static object file.  This is a form of emulation to support existing build systems.  If you want to build a runtime shared library use the SIDE_MODULE setting.')
-        link_to_object = True
 
     if shared.Settings.SUPPORT_BIG_ENDIAN:
       shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += [
-- 
2.25.1

